chainstate-storage/src/internal/mod.rs

```diff
@@ -374,19 +374,19 @@ impl<'a, Tx: traits::GetMapMut<'a, Schema>> StoreTx<Tx> {
 impl<T: traits::TransactionRw<Error = storage::Error>> traits::TransactionRw for StoreTx<T> {
     type Error = crate::Error;
 
-    fn commit(self) -> crate::Result<()> {
+    fn commit(&mut self) -> crate::Result<()> {
         self.0.commit().map_err(Into::into)
     }
 
-    fn abort(self) -> crate::Result<()> {
+    fn abort(&mut self) -> crate::Result<()> {
```

Comment:
I understand the motivation, but it seems logical for the methods such as `commit`, `abort` and `finalize` to consume the object.